woocommerce/woocommerce
# Blocks: Isolate performance tests from e2e tests

## Proposed changes

In this PR, we:
- isolate blocks performance tests from E2E tests, and
- fix the `pnpm test:performance` script to actually run them.

Since we migrated the blocks' performance tests to Playwright in #42943, they have been running along with our E2E tests in the CI, which happened because their filename fell under the `*.side_effects.block_theme.spec.ts` test match for the Side Effects group E2E runner. This was unintentional, as those tests were supposed only to be run locally for comparison purposes (see p1713545144131079/1713542886.203369-slack-C8X6Q7XQU). That PR also missed updating the script that runs those tests, so now instead of having to run explicitly the following as mentioned [here](https://github.com/woocommerce/woocommerce/pull/42943#:~:text=Run%20pnpm%20playwright%20test%20%2D%2Dconfig%3Dtests/e2e/playwright.side%2Deffects.config.ts%20%2D%2D%20plugins/woocommerce%2Dblocks/tests/e2e/tests/cart/cart%2Dblock.performace.block_theme.side_effects.spec.ts%20or%20use%20code%20editor%20UI%20to%20run%20the%20tests.):

```
pnpm playwright test --config=tests/e2e/playwright.side-effects.config.ts -- plugins/woocommerce-blocks/tests/e2e/tests/cart/cart-block.performace.block_theme.side_effects.spec.ts
```

…you can simply run:

```
pnpm test:performance cart-block
```

## Testing instructions

- Run `pnpm test:performance` locally and ensure all the tests pass.
- Ensure the measurements are still being saved to `plugins/woocommerce-blocks/tests/e2e/utils/performance/reports/e2e-cart-performance.json`
- Ensure E2E tests in CI aren't running the performance tests.
  _tip: There's only 1 suite (`cart-block`) that we're currently running within the first shard of the Side Effects group, so if it's not there, it means it's OK_ 😄
